### PR TITLE
implemented asynchronous version with tests

### PR DESCRIPTION
**Pull request by Pranavug**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,5 +58,13 @@ set_tests_properties(Test_2d
     PROPERTIES
     PASS_REGULAR_EXPRESSION "Tests Passed")
 set_tests_properties(Test_2d
+    PROPERTIES
+    FAIL_REGULAR_EXPRESSION "Tests Failed")
+
+add_test(NAME Test_2d_async COMMAND sh -c "./2d_nonlocal_async --test_batch < ../tests/2d_async.txt")
```
Comment (diehlpk) [inline CMakeLists.txt:64]: Can you please add some comments about the tests, e.g. what is tested and how do you test it?

File: src/2d_nonlocal_async.cpp [hunk 2]
```diff
@@ -9,23 +9,93 @@
...
+#include <fstream>
 
-#include "point.h"
-#include "print_time_results.hpp"
+#include "../include/point.h"
+#include "../include/print_time_results.hpp"
+#include "../include/writer.h"
 
 //mathematical constant PI
 #define PI 3.14159265
```
Comment (diehlpk) [inline src/2d_nonlocal_async.cpp:29]: Please use MP_I from the cmath header, instead of defining your own constant.

Comment (Pranavug) [reply src/2d_nonlocal_async.cpp]: Ok done. Please check latest commit

File: CMakeLists.txt (hunk 1 above)
Comment (Pranavug) [reply CMakeLists.txt:86]: Yes added, please check latest commit

Comment (diehlpk) [reply CMakeLists.txt:86]: LGMT!

**Pull request merged**